lib/nodejs/worker.ex

```diff
@@ -126,8 +133,14 @@ defmodule NodeJS.Worker do
     end
   end
 
+  defp reset_terminal(port) do
+    Port.command(port, "\x1b[0m\x1b[?7h\x1b[?25h\x1b[H\x1b[2J")
+    Port.command(port, "\x1b[!p\x1b[?47l")
```

Comment:
Can you explain why we still need this part if our port is no longer dumping any possibly strange characters into the running terminal? If we do need it, it'd be great to have a note with an explanation and/or reference about what it does and why.

Reply:
I removed the systematic escaping, as it seemed good enough while running iEX. But there is still a need for cleaning up the terminal output when closing the session, in case there are some escape characters that are not properly close. 
In short, this is close to running `stty sane` or `reset` in your terminal after shutting down iex. it makes sure there are no pending escape sequences in the tty output

Reply:
OK, I'm just a bit confused because I thought no output from the port should be hitting the terminal output anymore, including escape sequences or anything else.

Maybe I'd understand better if you showed a minimal example of a JS function that would mess up my terminal w/o this reset? I tried with the functions in the new tests (with a separate copy of each statement targeting stderr as well) and didn't see anything unusual in my iex terminal.

Reply:
It's not a just function. When you run the node process, it writes to std out, and uses some escape characters to go back to the start of the line, rewrite the previous row, that kind of things. This is what's messing up the terminal output, and corrupts it even after you terminate the iex process. 

This is fixing the problem while iex is running, and makes sure it restores the terminal in a clean state (like if you run the reset command) before terminating the process to make sure everything still looks good in your terminal

Reply:
@francois-codes whatever it is, can you share an example? In my mind, when the `node` process writes to stdout, that message should be consumed by the Elixir code and _not_ sent to the terminal ever, at all. Just like when you write `cat myfile.txt | grep whatever` you only see the output of `grep` in your terminal and not the whole file.

So far, I've been unable to write a JS file that breaks that mental model for me or understand conceptually why it's wrong, so I'm asking for the piece of evidence that unlocks that for me so I can confidently add this code into the library.

Reply:
Here's what happen. I have a single JS file with a function, I'm just running it with iex on main
https://github.com/user-attachments/assets/34d744e3-d121-4e0b-a3ff-8dbd61016ddb

Reply:
I see something similar on `master` as well, but not on _this branch_ without the `reset` stuff

Reply:
@grossvogel  this is what you can see in my video. After closing iex, you can see I run a mix command and the layout is broken. Then I run stty sane command (which is doing a similar reset) and things are back to normal

Reply:
Because both are needed:
- while the process is running to get a normal output while using iex
- when it's closing to make sure terminal remains clean after shutting down iex

I though you asked for a video that demonstrate why this is needed, this is why I did it on master so you can see the current problem.

Reply:
@francois-codes I'm sorry I'm having a hard time communicating here. I understand that you're saying it's needed, but I'd like to understand why, and for that I want an example of some code I can run that shows that if I use the code in this PR _but without the terminal reset code_, something bad will happen in my terminal.

Reply:
@grossvogel I thought I explained, but apparently I haven't been expressing myself clearly. 

But that's ok. We've been using a fork successfully for a couple weeks now since I pushed this, so if you don't feel good about having that in we'll continue using our fork instead and I will close this.

Reply:
@francois-codes to be clear, I have no problem with any of this code, but I don't want to put code into the library that's not necessary. I do 100% believe the other changes in this PR are necessary b/c I've been able to reproduce the problem they solve. Since adding those other changes, though, I'm unable to reproduce a problem solved by the reset logic specifically. My preferred outcome would be for you to show me that problem and we get this code into the library.

Maybe you can help me with _my_ reproduction steps. Here's what I'm doing:
- Create a new Phoenix app
- Add the NodeJS library that gets JS source from a file on my desktop
- That file contains 1 function that writes a bunch of escape sequences (from your tests) into stdout and stderr
- Call the function in that file using the NodeJS library in the app's PageController
- Start the app using `iex -S mix phx.server`
- Visit the home page over and over again
- Close the terminal
- My terminal seems fine

Where am I going wrong?

<details>

<summary>Details</summary>

![2024-12-04 12 52 30](https://github.com/user-attachments/assets/6b69ee61-a103-4fd6-9f13-82ff6855b35a)

<img width="553" alt="Screenshot 2024-12-04 at 12 53 49 PM" src="https://github.com/user-attachments/assets/3220a132-1e5c-4800-be33-d50e03c6fcd8">

<img width="555" alt="Screenshot 2024-12-04 at 12 54 00 PM" src="https://github.com/user-attachments/assets/383aaefc-145c-41e3-af30-e7d69c7b78df">

<img width="820" alt="Screenshot 2024-12-04 at 12 54 07 PM" src="https://github.com/user-attachments/assets/ee262559-2f64-4774-9094-6a7cb27c46e8">

</details>

Reply:
When you have the issue, is your terminal just fine before exit and then it gets messed up after? I was just really really hoping to get a reproduction, but ultimately I'd rather have the fix than have you run on your own fork

Reply:
@grossvogel if I might add my 2cents. 

I can't add any evidence because I personally never had that issue with messing up iex. Just, if that part (reset) doesn't hurt anyone, and it solves the problem for at least one person, I'd vote for including it 😉

Reply:
https://github.com/mrdotb/live_react/issues/10
While I did not experience my iex output getting corrupted  until exit. Could it be related to the terminal emulator / shell / zshrc config that can output some ansi code
francois is using vscode terminal + zsh while grossvogel is using iterm + zsh
Can you reproduce the iex getting corrupted on a default zsh session / another terminal emulator

Reply:
@francois-codes I'm so sorry for the back and forth and miscommunication, but I'm ready to merge this _with_ the reset changes if you'd be willing to revert that commit?

Reply:
I have the issue with iterm + zsh outside of vs-code, and with warp as well.
it may not affect all terminals, but it's not just mine. there's an issue I linked in the PR where this has been reported a while back already.

Yes sure, I haven't tried all terminals, shells and all possible scenarios, but all tests are passing and the shell corruption is gone everywhere I've tested